internal/peermanager/peerwantmanager.go

```diff
@@ -16,6 +16,18 @@ type Gauge interface {
 	Dec()
 }
 
+type wantType bool
+
+const (
+	wantTypeHave  wantType = false
+	wantTypeBlock wantType = true
+)
+
+type wantInfo struct {
+	sessions map[uint64]struct{}
+	tp       wantType
+}
+
```

Comment:
In the peerWantManager we now keep track of which sessions are interested in each want